i18n: initial refactor; add French strings (fixes #45)

Zopieux opened the pull request:
This fixes #45 and adds a base for adding other languages.

Please consider merging this *before* doing important refactors, as it adds many small changes (the `_()` calls) in most files, that are a PITA to track manually.

Zopieux commented:
pylint spits errors because I have to call a function before importing stuff, for the translations to work. Not sure what's the right workaround.

actionless commented on pikaur/main.py:
```diff
@@ -155,16 +159,16 @@ def cli_print_help(args):
         '\n',
         pacman_help,
-        '\n\nPikaur-specific options:\n' if pikaur_options_help else '',
+        '\n\n' + _('pikaur-specific options:') + '\n' if pikaur_options_help else '',
```
this line seems not correct

actionless commented on pikaur/i18n.py:
```diff
@@ -0,0 +1,11 @@
+_translation = None
+
+
+def load_translation():
```
you need to populate it on first call instead of expecting what somebody wil call this function

actionless replied:
for example you can just call that function right after defining

Zopieux commented:
@actionless Updated according to your remarks, thanks for being so quick! There were also missing translations (plurals), also fixed. Rebased on master.

I had to edit the Dockerfile, because currently the Travis CI *makepkg* uses master HEAD as source, even though we're in a PR (another repo or branch). So I force makepkg to use the current source directory by creating a tarball and using that.

actionless commented on maintenance_scripts/Dockerfile:
```diff
@@ -5,9 +5,11 @@ RUN useradd -m user && \
     echo "user ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
 
-RUN pacman -Syu --noconfirm --needed sudo binutils python-pylint
+RUN pacman -Syu --noconfirm --needed sudo binutils make python-pylint
```
i think it should go to `makedepends`

Zopieux replied:
I checked on the Arch docs/forums. make is part of base-devel which is assumed to be installed to use AUR package. So I guess this would go against the guidelines. Can add it anyway though, costs nothing.

actionless replied:
for sure it's better to adhere the guidelines

actionless replied:
mb then install here `base-devel` instead of make and binutils?

Zopieux replied:
Yeah I'll do that. For reference by the way:
> Note: The group [base-devel](https://www.archlinux.org/groups/x86_64/base-devel/) is assumed to be already installed when building with makepkg. Members of this group should not be included in makedepends array.
> — https://wiki.archlinux.org/index.php/PKGBUILD#makedepends

actionless commented on maintenance_scripts/Dockerfile:
```diff
@@ -5,9 +5,11 @@ RUN useradd -m user && \
 
 COPY . /opt/app-build/
 RUN chown -R user /opt/app-build/
+RUN sudo -u user tar --transform 's,^,pikaur-git/,' -cf pikaur-git.tar.gz . && \
```
also i think joining together lines 11-14 into one multiline RUN command will make build a little faster (especially locally)

Zopieux replied:
Okay will do.

Zopieux commented:
@actionless Updated!

actionless commented:
Thanks a lot for the big amount of changes!
